### PR TITLE
Feature/synchronized actions

### PR DESCRIPTION
**Pull request by modass**
Extension of the product-construction for ta by synchronized actions.

- non-synchronizing actions are handled as before
- for synchronizing actions it is needed to determine which automaton synchronizes on them to allow to synchronize several automata with different subsets of labels. Transitions are built incrementally component-wise.
- the automaton still may contain non-reachable components. This does not affect the analysis but the size of the automaton.

### CODE REVIEW

File: test/test_ta.cpp [hunk 1]
```diff
@@ -87,6 +87,117 @@ TEST_CASE("Comparison of TA configurations", "[ta]")
...
+{
+	CHECK(!(Transition(Location{"s0"}, "a", Location{"s0"})
+	        < Transition(Location{"s0"}, "a", Location{"s0"})));
+	// source
+	CHECK((!(std::string("s0") < "s1")
+	       || (Transition(Location{"s0"}, "a", Location{"s0"})
+	           < Transition(Location{"s1"}, "a", Location{"s0"}))));
+	CHECK((!(std::string("s0") > "s1")
+	       || (Transition(Location{"s1"}, "a", Location{"s0"})
+	           < Transition(Location{"s0"}, "a", Location{"s0"}))));
```
Comment (morxa) [inline test/test_ta.cpp:100]: As discussed, those can be simplified

**Review by morxa — COMMENTED** [review]
Looks very good! I have some minor comments

File: src/automata/include/automata/ta_product.hpp [hunk 2]
```diff
@@ -114,23 +130,103 @@ get_product(const std::vector<TimedAutomaton<LocationT, ActionT>> &automata,
...
+		  (synchronizing_actions.find(symbol) != std::end(synchronizing_actions));
+		for (const auto &[ta_i, ta] : ranges::views::enumerate(automata)) {
+			// before finalizing the construction for one automaton, we collect all candidates for
+			// synchronizing jumps to avoid expensive lookup.
+			std::map<ActionT, std::set<Transition<std::vector<LocationT>, ActionT>>>
+			  new_synchronizing_jumps;
+			for (const auto &[source_location, transition] : ta.get_transitions()) {
+				// Only consider transitions where the symbol matches. Note that if the automaton does not
+				// synchronize with this symbol, this check will always fail if the considered symbol is
+				// synchronizing.
```
[the remark this answers is not in the document]
Comment (modass) [reply src/automata/include/automata/ta_product.hpp:159]: This was meant as a minor comment on how to justify the statement in the else-branch where we definitely know, that the automaton synchronizes with this symbol - basically by having a jump that uses the symbol.

**Review by morxa — APPROVED** [review]
Looks good, thanks!

**Pull request merged**